web/client/utils/PluginsUtils.js

```diff
@@ -273,20 +337,24 @@ const PluginsUtils = {
         return {
             id: id || name,
             name,
-            impl: includeLoaded(name, loadedPlugins, impl.loadPlugin || impl.displayName || impl.prototype.isReactComponent ? impl : impl(stateSelector)),
+            impl: includeLoaded(name, loadedPlugins, getPluginImplementation(impl, stateSelector)),
             cfg: assign({}, impl.cfg || {}, isObject(pluginDef) ? parsePluginConfig(state, plugins.requires, pluginDef.cfg) : {}),
             items: getPluginItems(state, plugins, pluginsConfig, name, id, isDefault, loadedPlugins)
         };
     },
     getPluginItems,
-    getConfiguredPlugin: (pluginDef, loadedPlugins, loaderComponent) => {
+    getConfiguredPlugin: (pluginDef, loadedPlugins = {}, loaderComponent) => {
         if (pluginDef) {
-            const Plugin = loadedPlugins && loadedPlugins[pluginDef.name] || (pluginDef.plugin && !pluginDef.plugin.loadPlugin && pluginDef.plugin);
+            const impl = loadedPlugins[pluginDef.name] ||
+                !pluginDef.plugin.loadPlugin && pluginDef.plugin;
+            const id = isObject(pluginDef) ? pluginDef.id : null;
+            const stateSelector = isObject(pluginDef) ? pluginDef.stateSelector : id || undefined;
+            const Plugin = getPluginImplementation(impl, stateSelector);
             const result = (props) => {
                 return Plugin ? (<Plugin key={pluginDef.id}
-                    {...props} {...pluginDef.cfg} pluginCfg={pluginDef.cfg} />) : loaderComponent;
+                    {...props} {...pluginDef.cfg} pluginCfg={pluginDef.cfg} items={pluginDef.items || []}/>) : loaderComponent;
```

Comment:
while testing I noticed here a nested configuration
eg: 
```
{
    "name": "TOC",
    "priority": 1,
    "glyph": "1-layer",
    "position": 2,
    "size": 256,
    "tooltip": "Layers",
    "container": "left-menu",
    "cfg": {
        "name": "TOC",
        "cfg": {
            "activateMapTitle": false,
            "activateRemoveLayer": false,
            "activateSettingsTool": false,
            "activateTitleTooltip": false
        }
    },
    "items": []
}
```
so these props from cfg will not apply to Plugin `{...pluginDef.cfg} pluginCfg={pluginDef.cfg}`

Reply:
@allyoucanmap Good catch, I committed a fix for this